src/com/magento/idea/magento2plugin/inspections/xml/ObserverDeclarationInspection.java

```diff
@@ -0,0 +1,247 @@
+                        for (String moduleName: modulesWithSameObserverName.keySet()) {
+                            String problemKey = observerKey.concat("_").concat(moduleName);
+                            if (!eventProblems.containsKey(problemKey)){
+                                problemsHolder.registerProblem(
+                                    observerNameAttribute.getValueElement(),
+                                    String.format(
+                                        duplicatedObserverNameProblemDescription,
+                                        observerName,
+                                        eventNameAttributeValue,
+                                        moduleName,
+                                        modulesWithSameObserverName.get(moduleName)
+                                    ),
+                                    errorSeverity
+                                );
+                                eventProblems.put(problemKey, observerXmlTag);
+                            }
+                        }
+                    }
+                }
+            }
+
+            private HashMap<String, String> fetchModuleNamesWhereSameObserverNameUsed(String eventNameAttributeValue, String observerName, EventIndex eventIndex, PsiFile file) {
+                HashMap<String, String> modulesName = new HashMap<String, String>();
+                String currentFileDirectory = file.getContainingDirectory().toString();
+                String currentFileFullPath = currentFileDirectory.concat("/").concat(file.getName());
+                String scope = "global";
+
+                Collection<PsiElement> indexedEvents = eventIndex.getEventElements(eventNameAttributeValue, GlobalSearchScope.getScopeRestrictedByFileTypes(
+                        GlobalSearchScope.allScope(file.getProject()),
+                        XmlFileType.INSTANCE
+                ));
+
+                for (PsiElement indexedEvent: indexedEvents) {
+                    PsiFile indexedAttributeParent = PsiTreeUtil.getTopmostParentOfType(indexedEvent, PsiFile.class);
+                    if (indexedAttributeParent == null) {
+                        continue;
+                    }
+
+                    String indexedEventAttributeValue = ((XmlAttributeValue) indexedEvent).getValue();
+                    if (!indexedEventAttributeValue.equals(eventNameAttributeValue)) {
+                        continue;
+                    }
+
+                    String indexedFileDirectory = indexedAttributeParent.getContainingDirectory().toString();
+                    String indexedFileFullPath = indexedFileDirectory.concat("/").concat(indexedAttributeParent.getName());
+                    if (indexedFileFullPath.equals(currentFileFullPath)) {
+                        continue;
+                    }
+
```

Comment:
Hi @vasilii-b 
Could you please add all the areas (corn, webapis, etc.)?
https://devdocs.magento.com/guides/v2.3/architecture/archi_perspectives/components/modules/mod_and_areas.html

Thank you!